Copy edits for validation errors (#5777)
pydantic/pydantic: docs/usage/validation_errors.md

Pydantic attempts to provide useful validation errors. Below are details on common validation errors users
may encounter when working with pydantic, together with some suggestions on how to fix them.

## `arguments_type`

This error is raised when arguments passed to a function are not

> a `Tuple`, `List`, or dictionary.

Reviewer: ```suggestion
a tuple, list, or dictionary.
```
I think this is okay since these are terms of art in python world. I could possibly be convinced to use this instead:
```suggestion
a `tuple`, `list`, or `dict`.
```
but I think the original is better (minus the double 'a').

## `bool`

This error is raised when the value type is not valid for a Boolean field.

## `bool_parsing`

> This error is raised when the value is a string that is not valid for coercion to a Boolean.

Reviewer: ```suggestion
This error is raised when the value is a string that is not valid for coercion to a boolean.
```

I recognize it's subjective, but I don't think we need to capitalize the B in Boolean; for reference, the `mypy` docs don't: https://mypy.readthedocs.io/en/stable/builtin_types.html.

If you prefer to capitalize anyway, you can ignore this.

## `bytes_too_long`

> This error is raised when the `bytes` value length is greater than `Field.max_length`.

Reviewer: ```suggestion
This error is raised when the length of a `bytes` value is greater than `Field.max_length`.
```

## `bytes_too_short`

> This error is raised when the `bytes` value length is less than `Field.min_length`.

Reviewer: ```suggestion
This error is raised when the length of a `bytes` value is less than `Field.min_length`.
```

## `bytes_type`

> This error is raised when the value type is not valid for a `bytes` field.

Reviewer: ```suggestion
This error is raised when the type of an input value is not valid for a `bytes` field.
```

## `date_from_datetime_inexact`

This error is raised when the `datetime` value provided for a `date` field
has a nonzero time component.
For a timestamp to parse into a field of type `date`, the time components

> must be all zero.

Reviewer: ```suggestion
must all be zero.
```
? Can stay as is if I'm just splitting hairs.

## `datetime_object_invalid`

> This error is raised when the `datetime` object is not valid.

Reviewer: ```suggestion
This error is raised when something about the `datetime` object is not valid.
```
This is sort of weird, I think the tweaked phrasing helps clarify that it's intended as a bit of a catch-all?

(The example below demonstrates it can be raised for invalid timezones, for example)

## `extra_forbidden`

> This error is raised when the input values contain extra fields and `config.extra=forbid`.

Reviewer: ```suggestion
This error is raised when the input values contain extra fields and `model_config['extra']=forbid`.
```

## `greater_than_equal`

> This error is raised when the value is not greater than or equal to `Field.ge`.

Reviewer: ```suggestion
This error is raised when the value is not greater than or equal to a specified constraint.
```

## `invalid_key`

> This error is raised when the key type is not valid.

Reviewer: ```suggestion
This error is raised when the type of a `dict` key is not valid.
```

## `json_type`

> This error is raised when the input value type is not a JSON string.

Reviewer: ```suggestion
This error is raised when the input value is of a type that cannot be parsed as JSON.
```

## `less_than_equal`

> This error is raised when the value is not less than or equal to `Field.le`.

Reviewer: ```suggestion
This error is raised when the value is not less than or equal to the specified constraint.
```

## `list_type`

> This error is raised when the input value is not a `List`.

Reviewer: ```suggestion
This error is raised when the input value is not a `list`.
```

## `model_class_type`

This error is raised when you validate with `strict=True` and

> input value is not an instance of the model.

Reviewer: ```suggestion
the input value is not an instance of the model.
```

## `multiple_of`

> This error is raised when the input is not a multiple of `Field.multiple_of`.

Reviewer: ```suggestion
This error is raised when the input is not a multiple of `FieldInfo.multiple_of`.
```
or
```suggestion
This error is raised when the input is not a multiple of the specified value.
```

## `no_such_attribute`

> This error is raised when you assign value to an attribute that does not exist.

Reviewer: ```suggestion
This error is raised when you assign a value to an attribute that does not exist.
```

## `recursion_loop`

> This error is raised when a cyclic reference is detected.

Reviewer: ```suggestion
This error is raised when a cyclic reference is detected during serialization.
```
I believe this just applies during serialization..

## `set_type`

> This error is raised when the value type is not `Set`.

Reviewer: ```suggestion
This error is raised when the value type is not `set`.
```

## `string_type`

> This error is raised when the value type is not a string.

Reviewer: ```suggestion
This error is raised when the value is not a string.
```
(You can say the type is string/`str`, or the value is _a_ string. But I wouldn't say the "value type is a string")

## `tuple_type`

> This error is raised when the value type is not valid for a `Tuple` field.

Reviewer: ```suggestion
This error is raised when the value type is not valid for a `tuple` field.
```

## `union_tag_not_found`

> This error is raised when it is not possible to extract the tag using a discriminator.

Reviewer: ```suggestion
This error is raised when it is not possible to extract a tag using the discriminator.
```

## `url_scheme`

> This error is raised when the URL schema is not valid for the URL type.

Reviewer: ```suggestion
This error is raised when the URL scheme is not valid for the URL type of the field.
```

## `url_syntax_violation`

> This error is raised when the RUL syntax is not valid.

Reviewer: ```suggestion
This error is raised when the URL syntax is not valid.
```

## `url_too_long`

> This error is raised when the URL length is greater than 2083.

Reviewer: It's weird to me that this is a limitation.. but that's not a concern here.
